### PR TITLE
Convert to TypeScript

### PR DESCRIPTION
**Pull request by PedroHLC**

### CODE REVIEW

**Review by miguelcoba — APPROVED** [review]

File: lib/index.ts [hunk 1]
```diff
@@ -47,14 +48,12 @@ function main () {
...
-function die (explanation) {
+function die (explanation: string) {
     console.log(explanation);
     process.exit(1);
 }
 
-module.exports = main
-
-function buildTypes (data) {
+function buildTypes (data: JSON) {
```
Comment (renanpvaz) [inline lib/index.ts:56]: functions are missing their return types, even if its just `void`

File: lib/index.ts [hunk 2]
```diff
@@ -9,10 +10,10 @@ var moduleNamespace = 'I18n';
 var destNamespacePath = path.join(destPath, moduleNamespace);
 const configJson = path.join(projectPath, 'i18n.json')
 
-function main () {
+export function main () {
     if (fs.existsSync(configJson)) {
         const rawJSON = fs.readFileSync(configJson);
-        const json = JSON.parse(rawJSON);
+        const json = JSON.parse(rawJSON.toString());
```
Comment (renanpvaz) [inline lib/index.ts:16]: this json could have an expected type like 
```ts
Partial<{ source: string, dest: string, namespace: string }>

File: lib/index.ts [hunk 3]
```diff
@@ -120,7 +124,12 @@ function buildLang (sourceFileName, data) {
         { stdio: [ 'pipe', 1, 2 ] }
     );
 
-    let buffer = subprocess.stdin;
+    if (subprocess.stdin === null) {
+        die('Unable to pipe to elm-format!'); 
+        return false;
+    }
+    
+    let buffer : Writable = subprocess.stdin;
```
Comment (renanpvaz) [inline lib/index.ts:132]: I think you don't need a space between the name and the colon, perhaps we should format the file with something like prettier

File: lib/index.ts (hunk 2 above)
Comment (PedroHLC) [reply lib/index.ts:16]: Thanks, this wasn't covered in my "Learn TS in 20 minutes" lessons :rofl:

File: lib/index.ts (hunk 3 above)
Comment (PedroHLC) [reply lib/index.ts:132]: Sure, but better to do in another PR.

File: lib/index.ts (hunk 1 above)
Comment (PedroHLC) [reply lib/index.ts:56]: Done

**Pull request merged**